### PR TITLE
📊  WHO WASH (Water, Sanitation and Hygiene) dataset

### PR DESCRIPTION
**Pull request by spoonerf**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new data processing step for harmonizing country data and calculating population statistics in the garden dataset.
- **Chores**
	- Updated VS Code settings for better file management and formatting.
- **Documentation**
	- Added a new dataset `who.csv` containing WHO/UNICEF JMP data on water, sanitation, and hygiene progress since 2000.

### CODE REVIEW

File: etl/steps/data/garden/wash/2024-01-06/who.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+import re
+
+import numpy as np
+from owid.catalog import Table
+from owid.catalog import processing as pr
+from structlog import get_logger
+
+from etl.data_helpers import geo
+from etl.data_helpers.geo import add_regions_to_table
```
Comment (veronikasamborska1994) [inline etl/steps/data/garden/wash/2024-01-06/who.py:11]: don't think this is used anywhere..

File: etl/steps/data/garden/wash/2024-01-06/who.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+log = get_logger()
+
+REGIONS_TO_ADD = [
```
Comment (veronikasamborska1994) [inline etl/steps/data/garden/wash/2024-01-06/who.py:18]: also, maybe I am missing something but looks like you are not actually doing any regional aggregates.. maybe you used to but not anymore?

File: etl/steps/data/grapher/wash/2024-01-06/who.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load garden dataset.
+    ds_garden = paths.load_dataset("who")
+    tb = ds_garden["who"]
+    tb = tb.reset_index()
+    tb["year"] = tb["year"].astype("Int64")
```
Comment (veronikasamborska1994) [inline etl/steps/data/grapher/wash/2024-01-06/who.py:17]: not sure if this line is necessary? I think you can remove all of these: 
```
    tb = tb.reset_index()
    tb["year"] = tb["year"].astype("Int64")
    tb = tb.set_index(["country", "year", "residence"], verify_integrity=True)
```

File: etl/steps/data/garden/wash/2024-01-06/who.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
+        dest_dir, tables=[tb], check_variables_metadata=True, default_metadata=ds_meadow.metadata
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def drop_erroneous_rows(tb: Table) -> Table:
```
Comment (veronikasamborska1994) [inline etl/steps/data/garden/wash/2024-01-06/who.py:66]: looks like this function is no longer used unless I am missing something?

File: etl/steps/data/garden/wash/2024-01-06/who.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    """
+    columns = tb.columns.drop(["country", "year", "pop", "residence"])
+
+    for col in columns:
+        tb[f"{col}_pop"] = (tb[col] / 100) * tb["pop"]
+
+    return tb
+
+
+def clean_values(tb: Table) -> Table:
```
Comment (veronikasamborska1994) [inline etl/steps/data/garden/wash/2024-01-06/who.py:90]: also don't think you use this one anywhere?

File: etl/steps/data/garden/wash/2024-01-06/who.py [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+            "total_proportion_of_population_using_improved__sanitation_facilities__including_shared__sl",
+        ],
+        axis=1,
+        errors="ignore",  # not all tables have these columns
+    )
+
+    return tb
+
+
+def calculate_hygiene_no_services(tb: Table) -> Table:
```
Comment (veronikasamborska1994) [inline etl/steps/data/garden/wash/2024-01-06/who.py:119]: don't think this one is used anywhere either

File: etl/steps/data/garden/wash/2024-01-06/who.meta.yml [hunk 7]
```diff
@@ -0,0 +1,498 @@
...
+  residence_pop: <% if residence == "Rural" %>rural <% elif residence == "Urban" %>urban <% elif residence == "Total" %>total <%- endif -%>
+  hygiene_basic_description_short: &hygiene_basic_desc_short |
+    Share of the {definitions.residence} population with a handwashing facility with soap and water at home.
+  hygiene_limited_description_short: &hygiene_limited_desc_short |
+    Share of the {definitions.residence} population with a handwashing facility lacking soap and/or water at home.
+  hygiene_no_facility_description_short: &hygiene_no_facility_desc_short |
+    Share of the {definitions.residence} population with no handwashing facility at home.
+  sanitation_basal_description_short: &sanitation_basal_desc_short |
+    Share of the {definitions.residence} population using at least improved sanitation facilities that are not shared with other households. Improved sanitation facilities are those designed to hygienically separate excreta from human contact, and include: flush/pour flush toilets connected to piped sewer systems, septic tanks or pit latrines; pit latrines with slabs (including ventilated pit latrines), and composting toilets.
```
Comment (veronikasamborska1994) [inline etl/steps/data/garden/wash/2024-01-06/who.meta.yml:22]: this is up to you but I think the titles of these indicators will likely be self-explanatory and you don't need to repeat the "`Share of the {definitions.residence} population using at least improved sanitation facilities that are not shared with other households.`". Would make the subtitles a bit shorter :)

File: .vscode/settings.json [hunk 8]
```diff
@@ -6,7 +6,10 @@
   "files.trimTrailingWhitespace": true,
```
Comment (veronikasamborska1994) [inline .vscode/settings.json:6]: not sure if you intended to make changes to this file?

File: dag/wash.yml [hunk 9]
```diff
@@ -0,0 +1,11 @@
+steps:
+  data://meadow/wash/2024-01-06/who:
+    - snapshot://wash/2024-02-04/who.csv
+    - snapshot://wash/2024-02-04/who_regions.csv
+  data://garden/wash/2024-01-06/who:
+    - data://meadow/wash/2024-01-06/who
```
Comment (veronikasamborska1994) [inline dag/wash.yml:6]: just for consistency maybe it'd make sense to just have the same date across all steps? ie 2024-02-04

**Review by veronikasamborska1994 — APPROVED** [review]
just a couple of small things!

File: etl/steps/data/garden/wash/2024-01-06/who.py (hunk 2 above)
Comment (spoonerf) [reply etl/steps/data/garden/wash/2024-01-06/who.py]: Yeh I was going to but I didn't in the end, the guy I was talking to at the data provider sent me some regional aggregates, but forgot to remove this - thanks!

File: .vscode/settings.json (hunk 8 above)
Comment (spoonerf) [reply .vscode/settings.json:6]: Yeh I was getting this weird thing where my VScode was adding a '?' to the beginning of long lines in metadata.yml files, and this seemed to be the only way to stop it. I am pretty sure that's the only change and then the rest of the diff is slight formatting changes.

**Pull request merged**